src/socket.rs

```diff
@@ -332,19 +332,16 @@ impl NlSocket {
             mem.truncate(mem_read as usize);
             self.buffer = Some(StreamReadBuffer::new(mem));
         }
-        let msg = match self.buffer {
-            Some(ref mut b) => Nlmsghdr::deserialize(b)?,
-            None => unreachable!(),
-        };
+        let msg = Nlmsghdr::deserialize(self.buffer.as_mut().unwrap())?;
```

Comment:
I'd prefer to revert this for readability purposes.

Reply:
OK, done.